docs/support-matrix.md

```diff
@@ -0,0 +1,51 @@
+# Compatible Storage Providers
+
+Ark is compatible with a variety of storage providers for backup data, as well as for volume snapshots using either a plugin for the volume provider, or [restic][4] for filesystem level backups.
```

Comment:
Let's try introducing the topic a bit differently. This sentence/paragraph tries to pack too much information into too few words, but unpacking it doesn't add much value. Can we simplify down to something like:

"Ark supports a variety of storage providers for different backup and snapshot operations. As of version 0.6.0, a plugin system allows anyone ..." (and continue with rest of following paragraph)